### DISCUSSION TITLE
Crane for multi-interdependent compilation steps

### DISCUSSION THREAD

**Discussion by lcnbr** [ipetkov/crane#566] [General]
I am working on a cli application that uses rust as a backend and python for some frontend stuff and handling the interface.

The compilation is quite convoluted and the final outcome is a python script that should be callable from the terminal. The backend first needs to be compiled with maturin so that the python side can call it. Some packaged c dependencies must also be compiled. Additionally another maturin built pyO3 library that is a dependency of both the rust and the python side needs to be compiled. 

I am wondering if crane is the right tool for the job. Additionally a collaborator has made a shell script compiles all this correctly. Should I instead package this up inside a nix file?

**Comment by dpc** [comment]
crane shines if complex setup, IMO. But You it might require both cargo and Nix knowledge to get right.
